nixos/doc/manual/configuration/luks-file-systems.section.md

```diff
@@ -75,3 +79,37 @@ as [Trezor](https://trezor.io/).
 ```nix
 boot.initrd.luks.devices."/dev/sda2".fido2.passwordLess = true;
 ```
+
+### systemd Stage 1 {#sec-luks-file-systems-fido2-systemd}
+
+If systemd stage 1 is enabled, it handles unlocking of LUKS-enrypted volumes
+during boot. The following example enables systemd stage1 and adds support for
+unlocking the existing LUKS2 volume `root` using any enrolled FIDO2 compatible
+tokens.
+
+```nix
+boot.initrd = {
+  luks.devices.root = {
+    crypttabExtraOpts = [ "fido2-device=auto" ];
+    device = "/dev/sda2";
+  };
```

Comment:
I did not have to add this bit.

Reply:
If I remove the `crypttabExtraOpts` from `nixosTests.systemd-initrd-luks-fido2`, the test fails. I believe the option value is required. Maybe some other option added this value inside your configuration? Also, the manpage of `systemd-crypttab` specifies that this option needs to be set in crypttab. I don't see any other matches for the option in nixpkgs. This suggests that the option is not set otherwise.

Reply:
Anyway it would probably be more convenient if the option could be set automatically by `boot.initrd.luks.fido2Support`.

Reply:
https://github.com/Kranzes/nix-config/commit/ecdcb3621521fbf00f0ce944fe3621ccaf4de980 It was in your config at some point. Do you still use FIDO2 on that device?

Reply:
Yep, I use it on 2 devices now, none of them have this option anymore.

Reply:
This is somewhat puzzling. I'd expect the test not to fail in this case.

The idea of keeping things simple is understandable. I'd like to stick to the official documentation for now. We can remove the part from the manual when we figure out why it works for you.